presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlockBuilder.java

```diff
@@ -504,6 +504,12 @@ private static int getHashPosition(Block keyBlock, int position, MethodHandle ke
             throw new RuntimeException(throwable);
         }
 
-        return (int) Math.floorMod(hashCode, hashTableSize);
+        return getMod(hashCode, hashTableSize);
+    }
+
+    // This function will reduce the hashcode to [0, hashTableSize) uniformly.
+    static int getMod(long hashcode, int hashTableSize)
```

Comment:
Note this is not doing modulo. What about all it `fastrange` as used in author's code: https://github.com/lemire/fastrange ?

Maybe add a comment 
```
Adapted from the fastrange algorithm: https://github.com/lemire/fastrange
```
?

Reply:
How about naming it `getHashPosition` like the caller's method?  

Also, the comment is great.

Reply:
@dain there was already getHashPosition() function in MapBlockBuilder which calls this function. It computes the 64 bit hash code and apply this function on it. Also this function is non-related to hashing but just can be used to reduce a hash value to a smaller range. How about getPositionInReducedRange?

Reply:
I was thinking about the name describing the use as opposed to the implementation.  In the case of this class, it is selecting the hash position based on the provided hash code and hash table size.  From the readers perspective I think the use name will be more helpful, where as I think the description of the implementation is better placed in a comment.  That said, I don't feel super strongly about this.

Reply:
Thanks @dain. I have updated the name to getReducedHashPosition(). So in MapBlockBuilder, getHashPosition() will call this new getReducedHashPosition(). Also added the explanation and reference in the comments.

Reply:
As suggested by @martint I renamed the method name to computePosition()